extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -122,25 +130,57 @@ class AmpStickyAd extends AMP.BaseElement {
         this.visible_ = true;
         this.element.classList.add('-amp-sticky-ad-visible');
         this.viewport_.addToFixedLayer(this.element);
-        this.updateInViewport(this.ad_, true);
-        this.scheduleLayout(this.ad_);
         // Add border-bottom to the body to compensate space that was taken
         // by sticky ad, so no content would be blocked by sticky ad unit.
         const borderBottom = this.element./*OK*/offsetHeight;
         this.viewport_.updatePaddingBottom(borderBottom);
         this.addCloseButton_();
-        timerFor(this.win).delay(() => {
-          // Unfortunately we don't really have a good way to measure how long it
-          // takes to load an ad, so we'll just pretend it takes 1 second for
-          // now.
-          this.vsync_.mutate(() => {
-            this.element.classList.add('amp-sticky-ad-loaded');
-          });
-        }, 1000);
+        this.scheduleLayoutForAd_();
       });
     }
   }
 
+  scheduleLayoutForAd_() {
+    if (this.ad_.isBuilt()) {
+      this.layoutAd_();
+    } else {
+      listenOnce(this.ad_, 'amp:built', () => {
+        this.layoutAd_();
+      });
+    }
+  }
+
+  layoutAd_() {
+    if (!this.ad_.isFirstLayoutCompleted()) {
+      this.updateInViewport(this.ad_, true);
```

Comment:
So, all steps are the same except for the last one: display vs delay. Right? If so, please factor out of `if`.